### DISCUSSION TITLE
Conditional Filters

### DISCUSSION THREAD

**Discussion by Khushli** [twilio-labs/paste#2017] [General] [answered]
Project 1 - Account Insights (console) 
<img width="1054" alt="Screen Shot 2021-11-08 at 7 34 40 PM" src="https://user-images.githubusercontent.com/87436655/140840270-f9ef7254-7504-4094-a8bd-652ba674aac6.png">

Wanted to share the designs for conditional filters, we haven't found similar usecase for other Twilio products so we created our own keeping the Paste guidelines. 

Project 2 - Twizzy (internal tool)

<img width="583" alt="Screen Shot 2021-11-08 at 7 34 11 PM" src="https://user-images.githubusercontent.com/87436655/140840477-25bfc4b8-80dd-40f2-95cb-e3a389491bab.png">

Again, building filters using the disclosure component for filtering search data. Looking to get feedback on if we can use Paste to build this or any other suggestions to improve this experience.

**Comment by nkrantz** [comment, marked as answer]
Hi @Khushli! Just want to let you know that we've seen your post and we're discussing feedback as a team. @gloriliale is on Helper Duty and will respond with an update here soon. Thanks!

**Reply by gloriliale** [reply]
@Khushli so glad you got to discuss this w/ the design team! For reference, here's some conclusions:

- The implementation of the Project 1 - Accounts Insights (console) conditional filters seems to work fine. We should eventually include this example in the Filter pattern in the future when those docs are up.

- The Project 2 - Twizzy (internal tool) implementation of using the disclosure component for collecting multiple filters works. We suggested adding a number badge in the disclosures that house multiple options that could be selected to indicate how many filters were applied when the disclosure was closed

Thanks!

**Comment by kate-porter** [comment]
Hey @Khushli, I know we owe you a couple of answers regarding the Truncate component and the nested Checkboxes, too.

On the middle truncation, the Truncate component cannot do that out of the box, so we'd recommend truncating the end of the string unless you've uncovered a customer need for truncation in the center of the string. If you have, definitely let us know and we can chat about adding that functionality to the component.

On the checkboxes, I've just shared your designs with the team (apologies for the delay) and will get back to you with recommendations on how to build those nested checkboxes in an accessible way.

**Comment by kate-porter** [comment]
Hey @Khushli - I've finally got some info for you on your nested checkboxes.

It is possible to compose what you've got in your designs in an accessible way ([see these docs](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/checkbox/checkbox-2.html)). If you or your engineering partners need guidance from us when you get to implementation, please don't hesitate to reach out.

For the carets that expand/collapse the nested checkbox items, only clicking the icon itself will be able to expand/collapse the list, because clicking the checkbox label will check/uncheck the box. You'll want to make that expand/collapse icon a button (you can use our new [secondary icon button](https://paste.twilio.design/components/button#icon-buttons)). Also, I'd recommend using the `ChevronDisclosureCollapsed` and `ChevronDisclosureExpanded` icons for that button—we generally use the other chevron icon that you've used in your mocks to indicate that a UI element will pop out (such as a secondary menu, a dropdown, etc.).

Let me know if that all makes sense or if you have any other questions!

**Comment by Khushli** [comment]
Thank you all for all the helpful feedback. I will be coming to the OH once again to review all the updates I've made including how I've built multi-selection of account SIDs in a combobox without using pills.

**Comment by loreina** [comment]
Hi @Khushli - just wanted to share the office hours recording with you in case you'd like to revisit any of our conversations. [Here's the video](https://www.loom.com/share/e37ae97022b543b78ba79a3e70e09f14) (it includes timestamps for each topic), and the password is posted in the #help-design-system channel. 

To summarize some of our conversation:
- SIDs can be categorized by selected and the default list
- Success icon can be placed next to a selected SID, and removing the icon when a SID is deselected
- Using "..." to indicate the last page in the Pagination should be fine, but to explore with some user testing
- A chevron icon works to indicate a filter group that opens/closes

Thanks again for coming to office hours!
